pkg/auth/providers/oidc/oidc_provider.go

```diff
@@ -382,26 +370,17 @@ func (o *OpenIDCProvider) getUserInfo(ctx *context.Context, config *v32.OIDCConf
 		return userInfo, oauth2Token, err
 	}
 	oauthConfig := ConfigToOauthConfig(provider.Endpoint(), config)
+	var verifier = provider.Verifier(&oidc.Config{ClientID: config.ClientID})
 	if err := json.Unmarshal([]byte(authCode), &oauth2Token); err != nil {
 		oauth2Token, err = oauthConfig.Exchange(updatedContext, authCode, oauth2.SetAuthURLParam("scope", strings.Join(oauthConfig.Scopes, " ")))
 		if err != nil {
 			return userInfo, oauth2Token, err
 		}
-	}
-
-	rawToken, ok := oauth2Token.Extra("id_token").(string)
-	if !ok {
-		rawToken, ok = oauth2Token.Extra("access_token").(string)
-		if !ok {
+		_, err = verifier.Verify(updatedContext, oauth2Token.AccessToken)
+		if err != nil {
```

Comment:
Why are we replacing this code of reading oauth2Token.Extra("access_token").(string)? Was this oauth2Token.AccessToken always present?

Reply:
TLDR; It was alway present so it seems like a cleaner implementation

The example provided from the go-oidc library shows this: 

```
// Extract the ID Token from OAuth2 token.
    rawIDToken, ok := oauth2Token.Extra("id_token").(string)
    if !ok {
        // handle missing token
    }

    // Parse and verify ID Token payload.
    idToken, err := verifier.Verify(ctx, rawIDToken)
    if err != nil {
        // handle error
    }
```
which is what I did initially for the access token. But, as I was refactoring for nested groups, I noticed that if you look at the struct of the oauth2Token, it does have a field for AccessToken, pulling from the access_token json key: 

```
type Token struct {
	// AccessToken is the token that authorizes and authenticates
	// the requests.
	AccessToken string `json:"access_token"`

	// TokenType is the type of token.
	// The Type method returns either this or "Bearer", the default.
	TokenType string `json:"token_type,omitempty"`

	// RefreshToken is a token that's used by the application
	// (as opposed to the user) to refresh the access token
	// if it expires.
	RefreshToken string `json:"refresh_token,omitempty"`

	// Expiry is the optional expiration time of the access token.
	//
	// If zero, TokenSource implementations will reuse the same
	// token forever and RefreshToken or equivalent
	// mechanisms for that TokenSource will not be used.
	Expiry time.Time `json:"expiry,omitempty"`

	// raw optionally contains extra metadata from the server
	// when updating a token.
	raw interface{}
}
```
so I thought it was a cleaner implementation. We are also not using the id_token at any point, so I removed that check all-together.

Reply:
Thank you for the explanation :)